WordPress/wordcamp.org
# WCPT: Update build process, use React from `@wordpress/element`

Convert build process to use `wp-scripts`, and pull React dependencies from `@wordpress/element`. This will allow core to manage the vendor dependencies, enabling better caching, fewer conflicts, & a smaller overall filesize for `applications.min.js` (from 246K down to 9.3K). There should be no visible changes.

This also removes the one special-case project from our repo, everything uses the yarn workspaces now.

Fixes #317 

Note: This changeset is messy because each file had to be renamed from `*.jsx`, since wp-scripts doesn't understand how to process .jsx files. The basic overview of changes are:

- Create yarn workspace `wcpt`
- Rename files: `*.jsx -> *.js`
- Update PropTypes definitions to use the 3rd-party package ([removed from React core v15.5](https://reactjs.org/docs/typechecking-with-proptypes.html))
- Switch from `React.createClass` (also deprecated in v15.5) to native classes, extend Component from `@wordpress/element`
- Update [`createRef` usage,](https://reactjs.org/docs/refs-and-the-dom.html) previous version deprecated in v16.3 (or earlier?)
- Fix linter issues in JS & CSS
- Remove class properties ([this option](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties) is not supported in wp-scripts' process)
- In a few cases, moved from class components to functional components

### How to test the changes in this Pull Request:

1. Run `yarn workspace wcpt build`
2. Check that the application trackers for WordCamps & Meetups both still work
3. Tables are sortable & searchable, and show all expected content
